stdlib/private/TensorFlowUnittest/TensorFlowUnittest.swift

```diff
@@ -53,51 +53,34 @@ public func printT(_ message: String) {
 }
 
 extension TestSuite {
-  // Use the three macros CPU, CUDA and TPU to differentiate the 3 associated
-  // backends.
-  // Notes:
-  // 1. When Cuda is enabled, we still run testCPU(). This might change in the
-  // future, as the runtime based on TF C API will place nodes on GPU when GPU
-  // is available.
-  // 2. TPU is mutually exclusive with {CPU, CUDA} (requiring different
-  // hardware).
+  // If the macro TPU is not specified, run the tests with CPU or GPU. Otherwise
```

Comment:
It seems previously, CPU and GPU tests were both run with `-DCUDA`.
Is there a reason why now only GPU tests are run?

Reply:
This PR removes the runtime config distinction between CPU and GPU tests. At compile time, user uses -Xllvm -tf-target-gpu to specify whether they target CPU or GPU. The runtime is then agnostic of the CPU vs GPU device.

Unfortunately, it still needs to be aware of XLA/TPU for now. Revisiting that is outside the scope of this PR (I left a TODO there though).